### DISCUSSION TITLE
Translations

### DISCUSSION THREAD

**Discussion by realmazharhussain** [realmazharhussain/gdm-settings#23] [General]
I'm opening this discussion because I need a way to provide updates to translators of this application about new/modified text in the application so that they could translate it before new releases.

Some of my ideas are to use one of the following for translation updates:
- email newsletter
- announcements 1 week before a release
- this discussion

Any suggestions?

**Comment by daPhipz** [comment]
Great idea!
I think the easiest way would be to announce a pending release here and @ the people that have contributed translations in the past, if they agreed to be notified. Some projects I contribute to do it this way, and it works well IMO. What do you think, @Vistaus and @oscfdezdz?

Another idea would be to send an email to gnome-i18n@gnome.org, the mailing list for translation related things in the GNOME cosmos. Over at https://l10n.gnome.org/, there is infrastructure where all the "official" translation work for the core GNOME components happens. Maybe if you ask nicely, they incorporate your application in the [`gnome-extras`](https://l10n.gnome.org/releases/gnome-extras/) category? This would mean some work on your side to connect this repo to the Damned Lies infrastructure, though. I have no experience with that, unfortunately.

A third (but least favourite option IMO) would be to use an online translation service like [Weblate](https://hosted.weblate.org/). While these services tend to have many users, the translations are often low quality. This happens because the translators don't see the strings in context with the source code or in the UI - creating the potential for misunderstandings and other sources of translation errors.

My preferred way would be to get notified here, but I am open for other options as well - whatever floats your boat!

**Reply by oscfdezdz** [reply]
I agree, announcing a pending release a week or so in advance is the best way. We can subscribe to this discussion and be notified when the announcement happens.

**Reply by Vistaus** [reply]
I too agree. A notification on here a few days or a week in advance is the preferred option for me, but Weblate or Transifex would also be fine by me. GNOME Damned Lies would not be fine for me.

**Comment by realmazharhussain** [comment]
For now, it is fine for me to announce here a week before every release and mention every translator.

But if, some day, the number of translators grows up to something like 20 or more people, it would not be feasable for me to mention every single one of them.

We need a future proof way of doing this.

**Reply by daPhipz** [reply]
How about announcing it right here (or in a separate topic for each release)?

**Comment by realmazharhussain** [comment]
I've decided that I should post here 1 week before before each release. Anyone who wants to get updates can subscribe to this discussion.

Everyone onboard?

**Reply by Vistaus** [reply]
I am onboard. 🙂

**Reply by daPhipz** [reply]
Me too!

**Reply by oscfdezdz** [reply]
Yes!

**Comment by realmazharhussain** [comment]
Hey everyone, [translation instructions](https://github.com/realmazharhussain/gdm-settings/blob/main/docs/translate.md) have been updated. They have seen a few major changes.

- There's a new shell script `po/update-pot-file.sh` that creates `gdm-settings.pot` file
- A lot of new translatable strings will be detected now because of the new shell script `po/update-pot-file.sh`
- Suggested translation app is now [Poedit](https://poedit.net) instead of [GTranslator](https://gitlab.gnome.org/gnome/gtranslator.git).

**Comment by realmazharhussain** [comment]
### Release Announcement!

**Expected Release Data:** 22 April, 2022

The application is going into UI freeze starting now. The next version should be released next Friday (22 April). Update any translations before then.

**Reply by daPhipz** [reply]
I get the following error when trying to create the `pot` file:

```
po/update-pot-file.sh: 5: Syntax error: "(" unexpected
```

Can you reproduce this? Unfortunately, I have no idea how to fix it...

**Reply by realmazharhussain** [reply]
I have fixed it now. `git pull` and run `po/update-pot-file.sh` again.

**Comment by yarons** [comment]
Hey, how about an open source solution such as Weblate?

**Reply by realmazharhussain** [reply]
@daPhipz @Vistaus @oscfdezdz @Mejans @eshagh79 @rene-coty

Guys, you there?
We need your input on this.

**Reply by daPhipz** [reply]
> A third (but least favourite option IMO) would be to use an online translation service like Weblate. While these services tend to have many users, the translations are often low quality. This happens because the translators don't see the strings in context with the source code or in the UI - creating the potential for misunderstandings and other sources of translation errors.

**Reply by yarons** [reply]
@daPhipz You are aware of the facts that there's a screenshot support (which is a feature I helped promoting) and how is it any different from editing the po directly?

Using a web localization platform doesn't guarantee good translations but there are many tools for context and collaboration which are easier than implementing them directly on a po fie, and if the configuration is correct all the strings have links to their original location in the source code.

So it's not about bad platform, it's about bad translators or misuse of the localization platform, keeping the current status is mostly time consuming and inconvenient, we can make Weblate work best for us and I know this platform very well, we can configure it to increase context awareness and convenience.

The other localization platform often lacks some of these features so I don't mind avoiding them all at once.

**Reply by daPhipz** [reply]
In fact, I didn't know that - thanks for the heads-up! Ultimately, it is up to @realmazharhussain to decide, but I guess Weblate is at least worth a shot then ;)

**Reply by eshagh79** [reply]
I agree with Weblate, it is still better than manual translation in the worst case.

**Reply by realmazharhussain** [reply]
I'll add this project to Weblate.

**Comment by realmazharhussain** [comment]
## Release Announcement!

**Expected Release Date**: 20 June, 2022

The application is going into UI freeze starting now. The next version should be released next Monday (20 June). Update any translations before then.

**Reply by daPhipz** [reply]
:heavy_check_mark: German

**Reply by yarons** [reply]
✔️ Hebrew

**Reply by Mejans** [reply]
✅ Occitan

**Comment by realmazharhussain** [comment]
## Release Announcement!
Version 1.0 of the app is in beta now. There will be no change in translatable text in the app unless it is for a bug fix.

The release date is currently not set. However, v1.0 is expected to stay in beta for at least a month before it is released. Maybe, I could just release it when GNOME 43 is released. We'll see.

**Edit:** All Weblate translations are committed to `l10n` branch and not `main` branch. They will be merged into `main` at some point in the future.

**Edit:** Sorry! Descriptions of changes and fixes (which require translation) were needed to be added to the application's metainfo even though I said there will be no change in translatable text. Now, it seems like there will be no more changes in the app altogether (text or otherwise) before v1.0 is released (unless, of course, there is a bug to be squashed).

**Reply by daPhipz** [reply]
I hope that I did everything right and that the German translation is now almost done – could you please check? So far, nothing is committed yet, which makes me a bit nervous...

**Reply by realmazharhussain** [reply]
Sorry! I should have mentioned earlier; Translations are committed to `l10n` branch now.

This was done because daily translations were spamming the commit history. Now all the translation changes will be squash merged to `main` as a single commit, periodically.

**Reply by daPhipz** [reply]
Ah I see, good idea xD

**Comment by realmazharhussain** [comment]
## Release Announcement

**Version:** 1.0 (stable)
**Expected Release Date:** 23 September, 2022

It seems like this app is in good enough state to release as a stable version. So, the plan is to release it on 23st of September. If you would like your translations to be included in v1.0, please, update/add translations **before** September 23 (Friday).

**Comment by realmazharhussain** [comment]
## Discussion Announcement

Hi, future release announcements and translation-related updates will not be posted here. Instead, they will be posted directly to [Weblate](https://hosted.weblate.org/projects/gdm-settings).

**Reply by yarons** [reply]
This link redirects to an error page

**Reply by realmazharhussain** [reply]
Sorry! Fixed it now.
